scrapy/core/downloader/contextfactory.py

```diff
@@ -17,4 +23,6 @@ def getContext(self, hostname=None, port=None):
         # Enable all workarounds to SSL bugs as documented by
         # http://www.openssl.org/docs/ssl/SSL_CTX_set_options.html
         ctx.set_options(SSL.OP_ALL)
+        if hostname and ClientTLSOptions is not None: # workaround for TLS SNI
+            ClientTLSOptions(hostname, ctx)
```

Comment:
I'm not sure but I *think* this should be applied only to TLS connections. It would be good to test it against different versions of SSL and TLS protocols to make sure it works as expected in all the cases.